### ISSUE TITLE
EntityCollection is returned for empty results

### ISSUE THREAD

**Issue by TobiasGraml11** [vienthuong/shopware-php-sdk#58]
When using the repositories, I can't have explicit typehints, as the general one is returned for empty results. Would be great, if the correct collection class would be returned for the corresponding repository.

Code:
```php
$criteria1 = new Criteria();
$criteria1->setIds(['ca9f2ea7166c4ff58f42c2e100ca3fd5']);

$criteria2 = new Criteria();
$criteria2->setIds(['0134e8fa91da44c7b5418806d7b95c21']);

/** @var CategoryCollection $categories */
$categories1 = $this->categoryRepository
    ->search($criteria1, $this->context)
    ->getEntities();

$categories2 = $this->categoryRepository
    ->search($criteria2, $this->context)
    ->getEntities();
```

Result:
![image](https://user-images.githubusercontent.com/51320851/201676440-37ec0254-47df-4d44-9b31-140aab1bdc35.png)

**Comment by vienthuong** [comment]
Hi @TobiasGraml11 
Looks like a bug, I will have a look

**Comment by vienthuong** [comment]
I fixed this in latest 1.7.2 version, feel free to reopen if you have any issue with it
